src/main/c/Region.cpp

```diff
@@ -0,0 +1,56 @@
+#include "hail/Region.h"
+#include "hail/NativeObj.h"
+#include "hail/NativePtr.h"
+#include <jni.h>
+
+namespace hail {
+
+#define REGIONMETHOD(rtype, scala_class, scala_method) \
+  extern "C" __attribute__((visibility("default"))) \
+    rtype Java_is_hail_annotations_##scala_class##_##scala_method
+
+REGIONMETHOD(void, Region, nativeCtor)(
+  JNIEnv* env,
+  jobject thisJ
+) {
+  NativeObjPtr ptr = std::make_shared<Region>();
+  init_NativePtr(env, thisJ, &ptr);
+}
+
+REGIONMETHOD(void, Region, clearButKeepMem)(
+  JNIEnv* env,
+  jobject thisJ
+) {
+  auto r = static_cast<Region*>(get_from_NativePtr(env, thisJ));
+  r->clear_but_keep_mem();
+}
+
+REGIONMETHOD(void, Region, nativeAlign)(
+  JNIEnv* env,
+  jobject thisJ,
+  jlong a
+) {
+  auto r = static_cast<Region*>(get_from_NativePtr(env, thisJ));
+  r->align(a);
+}
+
+REGIONMETHOD(long, Region, nativeAlignAllocate)(
```

Comment:
No, on 64bit architectures it's conventional (universal across all implementations I've seen,
i.e. SPARC/Solaris, x86_64/Linux, x86_64/MS Visual C++) for "long" to be 64bits. I guess we could use int64_t, but I'm not a huge fan of using "Long" on the Scala side, "jlong" in the
JNI interfaces, and then insisting on calling it "int64_t" in C++, when "long" is in practice
the same thing.

And JNI has already forced us out of the realm of totally-standard-portable-C++ by
using "Long" as the type for virtual addresses - in C++ there's "intptr_t", meaning
"an integer type that's wide enough to hold a virtual address", which could be wider than
either "long" or "int64_t".

The motivation for all these corners of C++ standard types is much more to do with
low-end/embedded platforms.  It has been a long while since anyone had a workstation
or server cpu with anything other than 32bit long and void*, or 64bit long and void*.

Reply:
Okay, I see your point about using long for consistency with java and JNI.

Reply:
I think it's inevitable that this layer of the code is going to be inelegant and not an exemplar
of C++ style, since it's interacting closely with Scala and JNI.   What's important about the
foundation of a building is that it be solidly-constructed, not that it look good.  

If you want to change all the "long's" to "int64_t", I can do that if that gets us to the point where
we can commit this and move on to the more interesting code.   But I think we shouldn't be
going round in circles on these surface issues.